tests/SystemdFinderTest.php

```diff
@@ -0,0 +1,111 @@
+<?php
+declare(strict_types=1);
+
+namespace Setono\Deployer\Systemd;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Process\Process;
+
+final class SystemdFinderTest extends TestCase
+{
+    private const PATH = __DIR__ . '/systemd';
+    /**
+     * @test
+     */
+    public function it_finds_all(): void
+    {
+        $files = SystemdFinder::findAll(self::PATH, self::getCallback());
+
+        self::assertFiles([
+            'service1.service',
+            'service2---dev.service',
+            'service3---prod.service',
+            'service4---dev---r23.service',
+        ], $files);
+    }
+
+    /**
+     * @test
+     */
+    public function it_finds_dev_services(): void
+    {
+        $files = SystemdFinder::findByStage(self::PATH, 'dev', self::getCallback());
+
+        self::assertFiles([
+            'service2---dev.service',
+            'service4---dev---r23.service',
+        ], $files);
+    }
+
+    /**
+     * @test
+     */
+    public function it_finds_prod_services(): void
+    {
+        $files = SystemdFinder::findByStage(self::PATH, 'prod', self::getCallback());
+
+        self::assertFiles([
+            'service3---prod.service'
+        ], $files);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_find_other_stage(): void
+    {
+        $files = SystemdFinder::findByStage(self::PATH, 'staging', self::getCallback());
+        self::assertEmpty($files);
+    }
+
+    /**
+     * @test
+     */
+    public function it_finds_by_release(): void
+    {
+        $files = SystemdFinder::findByRelease(self::PATH, '23', self::getCallback());
```

Comment:
Release shouldn't be searchable without providing stage I believe.
I mean, if I searching release 12 - it can be either from dev or stage which is wrong.
So it should be like `findByStageAndRelease`

Reply:
Ah, we already have it...